spock-core/src/main/java/org/spockframework/runtime/GlobalExtensionRegistry.java

```diff
@@ -82,26 +79,61 @@ public Object getConfigurationByName(String name) {
   private void verifyGlobalExtension(Class<?> clazz) {
     if (!IGlobalExtension.class.isAssignableFrom(clazz))
       throw new ExtensionException(
-          "Class '%s' is not a valid global extension because it is not derived from '%s'"
+        "Class '%s' is not a valid global extension because it is not derived from '%s'"
       ).withArgs(clazz.getName(), IGlobalExtension.class.getName());
   }
 
-  private IGlobalExtension instantiateGlobalExtension(Class<?> clazz) {
+  private IGlobalExtension instantiateAndConfigureGlobalExtension(Class<?> clazz) {
```

Comment:
Wouldn't it make sense to have  `Class<IGlobalExtension>` all the way up to the source of those `Class` instances instead of `Class<?>` and then directly use the method below?

Reply:
Programming defensively, it makes sense to check the classes in `verifyGlobalExtension`.

Correct me if I'm wrong:
The problem of generics, we don't have `List<Class<IGlobalExtenstion>>`, but a list of `List<Class<? extends IGlobalExtenstion>>`, but you can't add anything to such a list, for that you'd need to declare it as `List<Class<? super IGlobalExtenstion>>`, but from that list you can't get anything useful from, so back to the unconstrained wildcard.